packages/passport-interface/README.md

```diff
@@ -1,3 +1,167 @@
+
+
+## 🛠 Install
+
+Install the `@pcd/passport-interface` package with npm:
+
+```bash
+npm i @pcd/passport-interface
+```
+
+or yarn:
+
+```bash
+yarn add @pcd/passport-interface
+```
+
+## 📜 Usage
+
+The communication between users and passport webapp is mostly via a popup, which redirects to the webapp according to the URL and its search parameters in the same popup. Once the webapp generates the requested content, the popup automatically closes, and the resulting content can be accessed through a React hook.
+
+For more detailed information about the package's functions and types, please refer to the [documentation site](https://docs.pcd.team/modules/_pcd_passport_interface.html).
+
+### Using the passport popup
+
+You can use the `usePassportPopupSetup` React hook to set up necessary passport popup logic on a specific route (e.g. `/popup`) and communicate with the passport webapp.
+
+```typescript
+import { usePassportPopupSetup } from "@pcd/passport-interface"
+
+export default function Popup() {
+    const error = usePassportPopupSetup()
+
+    return <div>{error}</div>
+}
+```
+
+Then the `usePassportPopupMessages` React hook can be used to process the results of your requests.
+
+```typescript
+import { usePassportPopupMessages } from "@pcd/passport-interface"
+
+export default function App() {
+    const [passportPCDString] = usePassportPopupMessages()
+
+     useEffect(() => {
+        console.log(passportPCDString)
+    }, [passportPCDString])
+}
+```
+
+### Adding a PCD to the passport
+
+You can generate a PCD and add it to the passport.
+
+```typescript
+import { EdDSAPCDPackage } from "@pcd/eddsa-pcd"
+import { constructPassportPcdProveAndAddRequestUrl, openPassportPopup } from "@pcd/passport-interface"
+import { ArgumentTypeName } from "@pcd/pcd-types"
+
+const url = constructPassportPcdProveAndAddRequestUrl<typeof EdDSAPCDPackage>(
+    process.env.PCDPASS_URL, // e.g. https://zupass.org.
+    window.location.origin + "/popup",
+    EdDSAPCDPackage.name,
+    {
+        id: {
+            argumentType: ArgumentTypeName.String
+    },
+        message: {
+            argumentType: ArgumentTypeName.StringArray,
+            value: ["0x342"]
+        },
+        privateKey: {
+            argumentType: ArgumentTypeName.String,
+            value: "0x42"
+        }
+    },
+    { title: "EdDSA Signature Proof" }
+)
+
+openPassportPopup("/popup", url)
+```
+
+Or add a previously generated serialized PCD.
+
+```typescript
+import { constructPassportPcdAddRequestUrl, openPassportPopup } from "@pcd/passport-interface"
+
+const url = constructPassportPcdAddRequestUrl(
+    process.env.PCDPASS_URL, // e.g. https://zupass.org.
+    window.location.origin + "/popup",
+    serializedPCD
```

Comment:
let's add some code to generate a simple serialized PCD so that this code snippet works as-is. Also, this is the sort of place where #355 is relevant.

Reply:
Are those functions (i.e. the ones in `PassportInterface.ts`) the only point where external devs can pass a wrong PCD name?